realm/realm-library/src/main/java/io/realm/rx/RealmObservableFactory.java

```diff
@@ -322,10 +322,12 @@ public void releaseReference(K object) {
             Integer count = references.get(object);
             if (count == null) {
                 throw new IllegalStateException("Object does not have any references: " + object);
-            } else if (count > 0) {
+            } else if (count > 1) {
                 references.put(object, count - 1);
-            } else {
+            } else if (count == 1) {
                 references.remove(object);
+            } else {
+                throw new IllegalStateException("Invalid reference count: " + count);
```

Comment:
I'm not entirely sure what happens if you call `subscription.unsubscribe()` twice though. Testing right now to verify.